src/sites/experiments.cubing.net/cubing.js/acube/index.ts

```diff
@@ -0,0 +1,203 @@
+import { puzzles, PuzzleLoader } from "../../../../cubing/puzzles";
+
+interface Piece {
+  [orientation: number]: Facelet;
+}
+type Mode = 'swap' | 'twist';
+
+class App {
+  mode: Mode = 'swap';
+  cube: Cube;
+
+  constructor() {
+    const puzzleSelect = document.querySelector(
+      "#puzzles",
+    ) as HTMLSelectElement;
+
+    for (const puzzle in puzzles) {
+      const option: HTMLOptionElement = puzzleSelect.appendChild(
+        document.createElement("option"),
+      )!;
+      option.value = puzzle;
+      option.textContent = puzzles[puzzle].fullName;
+    }
+
+    const puzzle = new URL(location.href).searchParams.get("puzzle") || '3x3x3';
+    if (puzzle) {
+      if (puzzle in puzzles) {
+        this.cube = new Cube(puzzles[puzzle]);
+        puzzleSelect.value = puzzle;
+      } else {
+        console.error("Invalid puzzle:", puzzle);
+      }
+    }
+
+    puzzleSelect?.addEventListener("change", () => {
+      this.cube.setPuzzle(puzzles[puzzleSelect.value]);
+
+      const url = new URL(location.href);
+      url.searchParams.set("puzzle", puzzleSelect.value);
+      window.history.replaceState("", "", url.toString());
+    });
+
+    document.querySelectorAll('input[name="mode"]').forEach((radio: HTMLInputElement) => {
+      radio.addEventListener('change', () => {
+        this.mode = radio.value as Mode;
+      });
+    });
+  }
+}
+
+class Cube {
+  selectedFacelet: Facelet | null;
+  pieces = new Map();
```

Comment:
this doesn't need to be a map really but it made typescript happy (was complaining about my indexes otherwise 🤷🏻‍♂️ )

Reply:
Looks like this is the main thing tripping up CI. I think a `Map` is a good choice here, although it needs to be `new Map<string, ____>()`

I haven't had time to check exactly what the underscore part would be, but if you pass it in (and remove extra semicolons) I think this file should pass the linter.